acceptancetests/assess_persistent_storage.py

```diff
@@ -223,6 +222,63 @@ def assess_charm_removal_single_block_and_filesystem_storage(client):
     else:
         log.info('Detach single block device storage output check: {}'.format(
             detach_single_blk_output_check))
+    # storage status change after remove-application takes some time.
+    # from experiments even 30 seconds is not enough.
+    time.sleep(60)
+    # check the real status of storage after remove-application
+    storage_output = json.loads(
+        client.get_juju_output('storage', '--format', 'json', include_e=False))
+    storage_list = storage_output['storage'].keys()
+    if len(storage_list) != 1:
+        log.error('\n'.join(storage_list))
+        raise JujuAssertionError(
+            'Unexpected number of storage unit(s). '\
+            'Found: {}\nExpected: 1\ntest terminated.'.format(
+            str(len(storage_list))))
+    elif single_fs_id in storage_list:
+        raise JujuAssertionError(
+            '{} should be removed along with remove-application.'.format(
+            single_fs_id))
+    elif single_blk_id not in storage_list:
+        raise JujuAssertionError(
+            '{} missing from storage list after remove-application.'.format(
+            single_blk_id))
+    else:
+        log.info(
+            'Check existence of persistent storage {} '\
+            'after remove-application: PASSED'.format(single_blk_id))
+    pool_storage = storage_output['volumes']['0']['storage']
+    storage_status = storage_output['volumes']['0']['status']['current']
+    if pool_storage != single_blk_id:
+        raise JujuAssertionError(
+            '{} missing from volumes.'.format(single_blk_id))
+    elif storage_status != 'detached':
+        raise JujuAssertionError(
+            'Incorrect status for {}. '\
```

Comment:
For all these status comparison blocks, I would expect to see a method that is reusable that you call, passing the expected string and the found status.